contents/blog/temporal-exports.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Why and how we moved exports to Temporal
+date: 2023-08-04
+author: ["tomas-farias", "ian-vanagas"]
+showTitle: true
+rootpage: /blog
+sidebar: Blog
+hideAnchor: true
+featuredImage: ../images/blog/posthog-engineering-blog.png
+featuredImageType: full
+category: Engineering
+---
+
+Letting users maintain ownership of their data is a key value of PostHog. Although we aspire to become the single source of truth for all your data needs, we realize we don’t fulfill everything everyone wants to do (yet). Also, we know that even if we were the single source of truth, users  still want to use their data on other platforms.
+
+Our export system provides the ability to maintain ownership of all of your data in PostHog and move it where you like. Unfortunately, it didn’t scale as well as we wanted. Luckily for you, we recently launched our [newly upgraded exports system](/docs/cdp/batch-exports), built using Temporal. This post covers why and how we made the change to Temporal.
+
+## The trouble with exports
+
+Although exports worked well most of the time, they were causing trouble for some customers. They were failing, erroring, and dropping data. Fixing these issues was taking up a significant amount of the pipeline team’s time. 
+
+Although stateless plugins (like GeoIP) worked well with our [plugin service structure](/blog/how-we-built-an-app-server), **stateful** plugins (like exports) didn’t. This is because they depended on external resources that can degrade or fail. Dealing with this required timeout and retry logic that was difficult  to build into the existing structure.
```

Comment:
nit: An extra space here:

```suggestion
Although stateless plugins (like GeoIP) worked well with our [plugin service structure](/blog/how-we-built-an-app-server), **stateful** plugins (like exports) didn’t. This is because they depended on external resources that can degrade or fail. Dealing with this required timeout and retry logic that was difficult to build into the existing structure.
```